File: app/src/main/java/com/rakuishi/ok/di/AppModule.java
package com.rakuishi.ok.di;

import android.app.Application;
import android.content.Context;

import com.google.gson.Gson;

import org.simpleframework.xml.core.Persister;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;

@Module
public class AppModule {

    private Context context;

    public AppModule(Application app) {
        this.context = app;
    }

    @Singleton
    @Provides
    public Context provideContext() {
        return context;
    }

    @Singleton
    @Provides
    public OkHttpClient provideHttpClient() {
        return new OkHttpClient();
    }

    @Singleton
    @Provides
    public Gson provideGson() {
        return new Gson();
    }

    @Singleton
    @Provides
    public Persister providePersister() {
        return new Persister();
    }
}

File: app/src/main/java/com/rakuishi/ok/di/AppModule.kt
package com.rakuishi.ok.di

import com.rakuishi.ok.data.FeedRepository
import com.rakuishi.ok.data.GitHubRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import javax.inject.Singleton

@InstallIn(ApplicationComponent::class)
@Module
class AppModule {

    @Singleton
    @Provides
    fun provideFeedRepository() = FeedRepository()

    @Singleton
    @Provides
    fun provideGitHubRepository() = GitHubRepository()
}